src/gui/XournalView.cpp

```diff
@@ -377,6 +384,19 @@ void XournalView::pageSelected(size_t page) {
     control->updatePageNumbers(currentPage, pdfPage);
 
     control->updateBackgroundSizeButton();
+
+    if (control->getSettings()->isEagerPageCleanup()) {
+        this->cleanupBufferCache();
```

Comment:
Please do not do this here. `getCurrentPage` is a method often used for input handling and should therefore be fast. What you do here is attaching a huge time penalty to each and every call of this method. Instead, introduce a background job that does this regularly.

Reply:
What does this have to do with `getCurrentPage`? Is this code not in `pageSelected` or am I missing something?

Reply:
Additionally, the enclosing function is doing file I/O which will take _far_ more cycles than garbage collecting the cache. I'm not sure whether this is going to have much effect in the grand scheme of things.

Reply:
The timer is insufficient for handling garbage collection, because rapidly scrolling through pages can cause entire RAM + swap to be consumed before the cache is GC'd.

Reply:
> What does this have to do with getCurrentPage? Is this code not in pageSelected or am I missing something?

Sorry my bad... Github folded the code with the header of getCurrentPage right above the fold and I didn't see the other method signature. In that case this is okay.